.github/workflows/build.yml

```diff
@@ -291,6 +292,19 @@ jobs:
           set +x
 
           cat report.html >> $GITHUB_STEP_SUMMARY
+
+          SUMMARY_SIZE=$(wc -c < $GITHUB_STEP_SUMMARY)
+          if [[ $SUMMARY_SIZE -gt 1000000 ]]; then
+            head -c 1000000 $GITHUB_STEP_SUMMARY | > $GITHUB_STEP_SUMMARY.tmp
+            mv $GITHUB_STEP_SUMMARY.tmp $GITHUB_STEP_SUMMARY
+            (
+              echo '.....'
+              echo ''
+              echo ':x: **WARNING: Summary is too large and has been truncated.**'
+              echo ''
+            )  >> $GITHUB_STEP_SUMMARY
+          fi
```

Comment:
There are a couple places where we add to $GITHUB_STEP_SUMMARY.

[Build](https://github.com/gitpod-io/gitpod/blob/3389607b52a9813e84ec8bd2492b791387ca7ff1/.github/workflows/build.yml#L294), which you limited to 1000000.

[DevBuild](https://github.com/gitpod-io/gitpod/blob/3389607b52a9813e84ec8bd2492b791387ca7ff1/.github/workflows/build.yml#L247), which I added a few months ago, but is not limited.

I'm not sure how much either is slowing down the build. I wanted to share though, because given a choice between filtering one or the other, my preference would be to filter DevBuild, and not Build.

Alternatively, we could filter both.

Either way, this feedback is not blocking, just a thought.

Reply:
@kylos101 I think we need to run it everywhere where we can hit the limit. I am not sure what's the value of having a summary that can trigger a failure because of an external limitation.
Also, what are we doing with the summary? If the intention is to see what's there, maybe we can store it in a bucket for a week?

Reply:
That is a good idea because it'll do GC.

If you're looking for alternatives (no GC):
https://github.com/marketplace/actions/append-to-gist
https://github.com/marketplace/actions/deploy-to-gist

:smiling_imp:

Reply:
Moved to another PR